pillar/kingfisher_main.sls

```diff
@@ -0,0 +1,201 @@
+        timeout: 300
+    rabbitmq:
+      configuration: rabbitmq
+      servername: rabbitmq.kingfisher.open-contracting.org
+
+postgres:
+  version: 15
+  # Public access allows Docker connections. Hetzner's firewall prevents non-local connections.
+  public_access: True
+  configuration:
+    name: kingfisher-main1
+    source: shared
+    context:
+      storage: ssd
+      type: oltp
+      content: |
+        # https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-MAX-WAL-SIZE
+        # https://github.com/open-contracting/deploy/issues/158
+        max_wal_size = 10GB
+
+        # https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-WAL-KEEP-SEGMENTS
+        # https://github.com/open-contracting/deploy/issues/158
+        wal_keep_size = 320
+
+        # https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-SYNCHRONOUS-COMMIT
+        synchronous_commit = local
+
+        # https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-SYNCHRONOUS-STANDBY-NAMES
+        synchronous_standby_names = 'pgslave001'
+
+        ### pgBackRest
+        # https://pgbackrest.org/user-guide.html#quickstart/configure-archiving
+
+        # https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-WAL-LEVEL
+        wal_level = logical
+
+        # https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-ARCHIVE-MODE
+        archive_mode = on
+
+        # https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-ARCHIVE-COMMAND
+        # https://pgbackrest.org/user-guide.html#async-archiving/async-archive-push
+        archive_command = 'pgbackrest --stanza=kingfisher-2023 archive-push %p'
+
+        # https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-MAX-WAL-SENDERS
+        max_wal_senders = 5
+  backup:
+    stanza: kingfisher-2023
+    retention_full: 4
+    s3_bucket: ocp-db-backup
+    s3_endpoint: s3.eu-central-1.amazonaws.com
+    s3_region: eu-central-1
+    repo_path: /kingfisher
+    configuration: kingfisher-main1
+    process_max: 4
+    # To check whether the error message in the grep command is up-to-date:
+    # "unable to remove file '%s'"
+    # https://github.com/pgbackrest/pgbackrest/blob/4adf6eed09da3f0819abef813c5a44deb9c91487/src/storage/storage.intern.h#L43
+    # "expire command encountered %u error(s), check the log file for details"
+    # https://github.com/pgbackrest/pgbackrest/blob/4adf6eed09da3f0819abef813c5a44deb9c91487/src/command/expire/expire.c#L1078
+    cron: |
+        MAILTO=root
+        # Daily incremental backup
+        15 05 * * 0-2,4-6 postgres pgbackrest backup --stanza=kingfisher-2023
+        # Weekly full backup
+        15 05 * * 3 postgres pgbackrest backup --stanza=kingfisher-2023 --type=full 2>&1 | grep -v "unable to remove file.*We encountered an internal error\. Please try again\.\|expire command encountered 1 error.s., check the log file for details"
```

Comment:
I added some comments above to check whether this grep still matches text.

I think the `1` needs to be changed to `\d+` (in which case `grep` needs a `-E` option)

I couldn't find what outputs `We encountered an internal error\. Please try again\.`

Is the `\|` meant to match a literal pipe character? (I'm not sure how pgbackrest concatenates messages).